### PR TITLE
Support adding vTPM device

### LINKED ISSUE #108: Feature request: Adding vTPM for Windows 11 during build ESXi 7.0.2
#### Description
Starting with build 22458 laste weekend Microsoft has started enforcing vTPM in virtual machines for Windows 11. So without a TPM it is not possible to install Windows 11.

#### Use Case(s)
All Windows 11 deployments need a TPM.  My workaround of power down VM, add vTPM in webgui, power on VM is not easily automated outside of packer, so it would be best to add the feature.

Windows Server 2022 currently does not need TPM, but future versions could.

#### Potential configuration

A new option in Hardware Configuration

vTPM (bool) - Enable vTPM for virtual machine. Defaults to false.

#### Potential References

Adding a vTPM from powerCLI can be done via New-VTpm -VM $VM

### PR DESCRIPTION
**Pull request by VladRassokhin**
Closes #108

### CODE REVIEW

File: builder/vsphere/common/step_hardware_test.go [hunk 1]
```diff
@@ -61,6 +61,39 @@ func TestHardwareConfig_Prepare(t *testing.T) {
...
+			expectedErrMsg: "'vTPM' could be enabled only when 'firmware' set to 'efi' or 'efi-secure'",
+		},
+		{
+			name: "Validate 'vTPM' and empty firmware",
+			config: &HardwareConfig{
+				VTPMEnabled: true,
+			},
+			fail:           true,
+			expectedErrMsg: "'vTPM' could be enabled only when 'firmware' set to 'efi' or 'efi-secure'",
+		},
```
Comment (azr) [inline builder/vsphere/common/step_hardware_test.go:96]: Nice !

File: builder/vsphere/driver/vm.go [hunk 2]
```diff
@@ -625,6 +626,28 @@ func (vm *VirtualMachineDriver) Configure(config *HardwareConfig) error {
...
+		return err
+	}
+	TPMs := devices.SelectByType((*types.VirtualTPM)(nil))
+	hasTPM := len(TPMs) > 0
+	if config.VTPMEnabled != hasTPM {
+		if !hasTPM {
+			device := &types.VirtualTPM{}
+			err = vm.addDevice(device)
+		} else {
+			err = vm.RemoveDevice(false, TPMs...)
```
Comment (azr) [inline builder/vsphere/driver/vm.go:644]: Is this important to add too ?

**Review by azr — APPROVED** [review]
Just had a question, other than that, this LGTM ! Thanks !

File: builder/vsphere/driver/vm.go (hunk 2 above)
Comment (VladRassokhin) [reply builder/vsphere/driver/vm.go:644]: AFAIR I've copy-pasted idea of removing existing TPM from somewhere else. Not sure whether if removing TPM should be supported. I'd vote for keeping that code.

Comment (azr) [reply builder/vsphere/driver/vm.go:644]: Cool, you probably have much more insight in this code area than me, let's keep this !

**Pull request merged**